File: examples/functions/next-style/users/_shared.js
function json(payload) {
  return {
    status: 200,
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload),
  };
}

function sharedMeta() {
  return {
    runtime: "node",
    helper: "users/_shared.js",
  };
}

function buildUsersIndexPayload() {
  return {
    route: "/users",
    ...sharedMeta(),
    users: [
      { id: "123", label: "Ada" },
      { id: "456", label: "Kai" },
    ],
  };
}

function buildUserDetailPayload(params) {
  return {
    route: "/users/:id",
    params: params || {},
    ...sharedMeta(),
  };
}

module.exports = {
  json,
  buildUsersIndexPayload,
  buildUserDetailPayload,
};

File: examples/functions/next-style/blog/_shared.py
import json


def respond(payload):
    return {
        "status": 200,
        "headers": {"Content-Type": "application/json"},
        "body": json.dumps(payload),
    }


def build_blog_payload(route, params=None, intro=None):
    params = dict(params or {})
    payload = {
        "route": route,
        "params": params,
        "runtime": "python",
        "helper": "blog/_shared.py",
    }
    slug = params.get("slug")
    if isinstance(slug, str) and slug:
        payload["slug"] = slug
    if intro:
        payload["intro"] = intro
    return respond(payload)
